zkvm/docs/zkvm-spec.md

```diff
@@ -1362,45 +1382,47 @@ _contract(P) proof prog_ **call** → _results..._
 7. Places the [payload](#contract-payload) on the stack (last item on top).
 8. Set the `prog` as current.
 
-Fails if the top two items are not [data](#data-type) or the third from top is not a [contract](#contract-type).
+Fails if:
+1. `prog` is not a [program type](#program-type),
```

Comment:
Same comment about "a program type," "a data type," and "a contract type." I'd much prefer "a program," "a data item," and "a contract."